Marketplace/AzureStack.Marketplace.psm1

```
@@ -0,0 +1,194 @@
+        [ValidateNotNullorEmpty()]
+        [String] $GalleryItemName
+    )
+
+    $subscription = (Get-AzureStackAdminSubscription -TenantId $tenantId -AzureStackCredentials $azureStackCredentials -ArmEndpoint $ArmEndpoint)
+
+    if((Get-Module AzureStack).Version -ge [System.Version] "1.2.9"){
+        Remove-AzureRMGalleryItem -Name $GalleryItemName
+    }else{
+        Remove-AzureRMGalleryItem -SubscriptionId $subscription -Name $GalleryItemName -ApiVersion '2015-04-01'
+    }
+}
+
+function Get-AzureStackAdminSubscription {
+    param (
+        [parameter(HelpMessage="Name of the Azure Stack Environment")]
+        [string] $EnvironmentName = "AzureStack",
+	
+        [parameter(mandatory=$true, HelpMessage="TenantID of Identity Tenant")]
+        [string] $tenantID,
+
+        [parameter(HelpMessage="Credentials to retrieve token header for")]
+        [System.Management.Automation.PSCredential] $azureStackCredentials,
+
+        [Parameter(HelpMessage="The administration ARM endpoint of the Azure Stack Environment")]
+        [string] $ArmEndpoint = 'https://adminmanagement.local.azurestack.external'
+    )
+
+    if(-not $azureStackCredentials){
+        $azureStackCredentials = Get-Credential
+    }
+    
+    if(!$ARMEndpoint.Contains('https://')){
+        if($ARMEndpoint.Contains('http://')){
+            $ARMEndpoint = $ARMEndpoint.Substring(7)
+            $ARMEndpoint = 'https://' + $ARMEndpoint
+        }else{
+            $ARMEndpoint = 'https://' + $ARMEndpoint
+        }
+    }
+    
+    $ArmEndpoint = $ArmEndpoint.TrimEnd("/")
+
+    try{
+        Invoke-RestMethod -Method Get -Uri "$($ARMEndpoint.ToString().TrimEnd('/'))/metadata/endpoints?api-version=2015-01-01" -ErrorAction Stop | Out-Null
+    }catch{
+        Write-Error "The specified ARM endpoint: $ArmEndpoint is not valid for this environment. Please make sure you are using the correct administrator ARM endpoint for this environment." -ErrorAction Stop
+    }
```

Do you want to rethrow the exception here?